.github/workflows/services-json.yml

```diff
@@ -0,0 +1,89 @@
+name: Services Validator
+
+on:
+  push:
+    paths:
+      - "plugins/rtmp-services/data/services.json"
+      - "plugins/rtmp-services/data/package.json"
+  pull_request:
+    paths:
+      - "plugins/rtmp-services/data/services.json"
+      - "plugins/rtmp-services/data/package.json"
+  schedule:
+  - cron: 0 0 * * *
+  workflow_dispatch:
+
+jobs:
+  schema:
+    name: Schema
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install & Configure Python
+        run: |
+          sudo apt install python3.9-dev
+          python3.9 -m pip install jsonschema json_source_map
+
+      - name: Validate Service JSON Schema
+        run: |
+           python3.9 CI/check-jsonschema.py plugins/rtmp-services/data/services.json plugins/rtmp-services/data/schema/service-schema-v3.json
+
+      - name: Annotate Errors
+        if: ${{ failure() && github.event_name == 'pull_request' }}
+        uses: yuzutech/annotations-action@v0.4.0
+        with:
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          title: 'Service JSON Errors'
+          input: './validation_errors.json'
+
+  service_check:
+    name: Service Check
+    runs-on: ubuntu-latest
+    needs: schema
+    # todo remove my repo before un-drafting PR
+    if: (github.repository_owner == 'obsproject' ||  github.repository_owner == 'derrod') && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch')
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Restore cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/other
+          key: service-check
+
+      - name: Install & Configure Python
+        run: |
+          sudo apt install python3.9
+          python3.9 -m pip install requests
+
+      - name: Check Services
+        id: check
+        run: |
+           python3.9 -u CI/check-services.py
+           # if there are unstaged changes, run the PR step
+           if [ -n "$(git status --porcelain=v1 2>/dev/null)" ]; then
```

Comment:
Use `if git diff --exit-code &>/dev/null; then` - no need to do string comparisons here.

Reply:
Seems like `git diff --quiet` would be an even better choice here, it implies `--exit-code` but silences all output so the redirect isn't required.